### ISSUE TITLE
关于feature 和 relation reuse 的疑惑

### ISSUE THREAD

**Issue by xingshulicc** [blackfeather-wang/Dynamic-Vision-Transformer#2]
1. 我们知道一个transformer应该是由多个encoder blocks组成的，那么我好奇的是upstream transformer 最后一层的输出是否要与downstream transformer每一个encoder block中的mlp输出进行级联？
2. 论文中提到要重用upstream transformer的attention logits, 也就是重用upstream transformer中由Q与K生成的attention map, 那么我所好奇的是，是不是要将upstream transformer 每一个encoder block中的 attention map都与 downstream transformer与之深度对应的encoder block的attention map 进行级联来达到relation resue的目的？
3. 这种重用机制所带来的额外计算开销理论上来说是非常巨大的，就像densenet的dense connection, 而论文中提到额外的计算开销是很小的，那么我觉得只有一个理由能解释这种相对额外开销很小的原因就是每一个patch 进行linear projection后得到的D的数值是很小的。我这样理解对吗？

**Comment by forever10086** [comment]
1. feature reuse是将最后一层的logit做不同的变换输入到对应的下层模型中去
2. relationship reuse是将上层模型的所有层的attention map拼接，展平，上采样，然后再做像feature reuse那样的变换，将变换后的结果在添加到下层模型的softmax中去
3. 确实这样来看，计算量也不会小
